cmd/ecs_task.go

```diff
@@ -4,53 +4,70 @@ import (
+	ecsTaskCmd.Flags().BoolVarP(&applyFlag, "apply", "a", false, "actually perform task definition deregistration")
+	ecsTaskCmd.Flags().IntVarP(&cutoffFlag, "cutoff", "c", 5, "how many most-recent task definitions to keep around")
+	ecsTaskCmd.Flags().BoolVarP(&debugFlag, "debug", "d", false, "enable for all the output")
+	ecsTaskCmd.Flags().IntVarP(&parallelFlag, "parallel", "p", 2, "how many concurrent deregistration requests to make")
+	ecsTaskCmd.Flags().BoolVarP(&quietFlag, "quiet", "q", false, "minimize output")
+	ecsTaskCmd.Flags().BoolVarP(&verboseFlag, "verbose", "v", false, "enable for chattier output")
+	rootCmd.AddCommand(ecsTaskCmd)
+}
+
 var ecsTaskCmd = &cobra.Command{
 	Use:   "ecs-task",
 	Short: "Deregister unused task definitions (dry run by default).",
 	Long: `Deregister unused task definitions (dry run by default).
 
 BEFORE RUNNING: Make sure that you've properly configured your environment with
-the AWS CLI for the AWS account you want to clean up.`,
+the following environment variables:
+
+AWS_ACCESS_KEY
+AWS_SECRET_ACCESS_KEY
+AWS_REGION`,
 	Run: func(cmd *cobra.Command, args []string) {
 		if parallelFlag < 1 {
-			fmt.Println("minimum parallel is 1")
+			fmt.Println("Minimum value for parallel flag is 1.")
 			os.Exit(1)
 		}
 
 		if debugFlag {
 			verboseFlag = true
 		}
 
-		flags := map[string]interface{}{
-			"apply":    applyFlag,
-			"cutoff":   cutoffFlag,
-			"debug":    debugFlag,
-			"parallel": parallelFlag,
-			"region":   regionFlag,
-			"verbose":  verboseFlag,
+		if quietFlag && verboseFlag {
+			fmt.Println("Can't set quiet flag alongside verbose or debug flags.")
+			os.Exit(1)
 		}
 
-		ecstask.Run(cmd, args, flags)
-	},
-}
+		ecsClient := ecsclient.NewECSClient()
 
-var applyFlag bool
-var cutoffFlag int
-var debugFlag bool
-var parallelFlag int
-var regionFlag string
-var verboseFlag bool
+		if err := ecsClient.ConfigureSession(); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
 
-func init() {
-	ecsTaskCmd.Flags().BoolVarP(&applyFlag, "apply", "a", false, "actually perform task definition deregistration")
-	ecsTaskCmd.Flags().IntVarP(&cutoffFlag, "cutoff", "c", 5, "how many most-recent task definitions to keep around")
-	ecsTaskCmd.Flags().BoolVarP(&debugFlag, "debug", "d", false, "enable for all the output")
-	ecsTaskCmd.Flags().IntVarP(&parallelFlag, "parallel", "p", 2, "how many concurrent deregistration requests to make")
-	ecsTaskCmd.Flags().StringVarP(&regionFlag, "region", "r", "us-west-2", "the AWS region in which to operate")
-	ecsTaskCmd.Flags().BoolVarP(&verboseFlag, "verbose", "v", false, "enable for chattier output")
-	rootCmd.AddCommand(ecsTaskCmd)
+		ecsClient.Flags.Apply = applyFlag
+		ecsClient.Flags.Cutoff = cutoffFlag
+		ecsClient.Flags.Debug = debugFlag
+		ecsClient.Flags.Parallel = parallelFlag
+		ecsClient.Flags.Quiet = quietFlag
+		ecsClient.Flags.Verbose = verboseFlag
+
+		if err := ecsClient.CleanupTaskDefinitions(); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
```

Comment:
Have you been able to confirm that the program cleanly exits on error here?

Reply:
Not through testing, but with the removal of concurrency this shouldn't be a concern anymore.